src/app/src/test/kotlin/org/vaccineimpact/orderlyweb/tests/unit_tests/controllers/api/ReportControllerTests.kt

```diff
@@ -125,6 +127,56 @@ class ReportControllerTests : ControllerTest()
                 .hasMessageContaining("*/reports.read")
     }
 
+    @Test
+    fun `getAllVersions returns all report versions if user has global read permissions`()
+    {
+        val now = Instant.now()
+        val reportVersions = listOf(
+                ReportVersion("r1", "display1", "v1", "v1", true, now,
+                        "auth", "req"),
+                ReportVersion("r2", "display2", "v2", "v2", true, now,
+                        "auth", "req")
+        )
+
+        val orderly = mock<OrderlyClient> {
+            on { this.getAllReportVersions() } doReturn reportVersions
+        }
+
+        val mockContext = mock<ActionContext> {
+            on { it.permissions } doReturn permissionSetGlobal
+        }
+
+        val sut = ReportController(mockContext, orderly, mock<ZipClient>(),
+                mock<OrderlyServerAPI>(),
+                mockConfig)
+
+        val result = sut.getAllVersions()
+        assertThat(result).hasSameElementsAs(reportVersions)
+    }
+
+    @Test
+    fun `getAllVersions throws MissingRequiredPermission error if user has no report reading permissions`()
+    {
+        val reports = listOf(Report(reportName, "test full name 1", "v1"),
```

Comment:
You don't actually need to mock the orderly call here as it will never get made. Can just do  
```
val sut = ReportController(mockContext, mock(), mock(), mock(), mockConfig)
```